src/renderer_d3d12.cpp

```diff
@@ -2952,7 +2952,7 @@ namespace bgfx { namespace d3d12
 
 		static void patchCb0(DxbcInstruction& _instruction, void* _userData)
 		{
-			union { void* ptr; uint32_t offset; } cast = { _userData };
+			const uint32_t offset = uint32_t(reinterpret_cast<uintptr_t>(_userData));
```

Comment:
You need to realize that I consciously didn't want `reinterpret_cast` when I wrote this code. I was aware that `reinterpret_cast` exist before...